jq/0002-Reject-NaN-with-payload-while-parsing-JSON.patch

```diff
@@ -0,0 +1,33 @@
+From 7a7cea283237cd24557c0b39954ebcd260802f34 Mon Sep 17 00:00:00 2001
+From: itchyny <itchyny@cybozu.co.jp>
+Date: Wed, 5 Mar 2025 07:43:54 +0900
+Subject: [PATCH 2/2] Reject NaN with payload while parsing JSON
+
+This commit drops support for parsing NaN with payload in JSON like
+`NaN123` and fixes CVE-2024-53427. Other JSON extensions like `NaN` and
+`Infinity` are still supported. Fixes #3023, fixes #3196, fixes #3246.
+
+(cherry picked from commit a09a4dfd55e6c24d04b35062ccfe4509748b1dd3)
```

Comment:
Would be useful to mention the dropped parts of the patch (the added tests) and why (because they don't apply cleanly) in the patch header here.

A nice feature to add to melange would be a filterdiff like capacity to the cherry-picks mechanism, that let you filter out a conflicting part of a commit (like an entry to a changelog, or, in this case, testcases that don't apply and that we wouldn't run :disappointed: ) ; basically a way to express the equivalent of:

```
git show a09a4dfd55e6c24d04b35062ccfe4509748b1dd3  | filterdiff -x '?/tests/*'
```

But maybe that lets one tweak a cherry-picked commit too much.

Reply:
yes, when we implemented the cherry-picks arg to the git-checkout pipeline, the very first thing we did was like "oh, but what i really want is filterdiff". I do wonder if we could do something like git format-patch; followed by filterdiff; followed by git am with extra comments / headers added on things that are filtered out. but yeah, that's a wishlist in the melange project.

Reply:
@stevebeattie wrt filterdiff - https://github.com/chainguard-dev/melange/issues/1422